macros/cargo

```diff
@@ -16,15 +16,45 @@
 %__cargo_cross_opts %{__cargo_common_opts} --target %{__cargo_target}
 %__cargo_cross_opts_static %{__cargo_common_opts} --target %{__cargo_target_static}
 %__cargo_env CARGO_TARGET_DIR="${HOME}/.cache" SKIP_README="true"
+%__cargo_env_fips CARGO_TARGET_DIR="${HOME}/.cache/.fips" SKIP_README="true"
+%__cargo_env_aws_sdk CARGO_TARGET_DIR="${HOME}/.cache/.aws-sdk" SKIP_README="true" RUSTFLAGS="%{__global_rustflags_aws_sdk}"
+%__cargo_env_fips_aws_sdk CARGO_TARGET_DIR="${HOME}/.cache/.fips/.aws-sdk" SKIP_README="true" RUSTFLAGS="%{__global_rustflags_aws_sdk}"
 %__cargo_env_static CARGO_TARGET_DIR="${HOME}/.cache/.static" SKIP_README="true"
+%__cargo_env_static_fips CARGO_TARGET_DIR="${HOME}/.cache/.fips/.static" SKIP_README="true"
 %__cargo_outdir "${HOME}/.cache/%{__cargo_target}/release"
+%__cargo_outdir_fips "${HOME}/.cache/.fips/%{__cargo_target}/release"
+%__cargo_outdir_aws_sdk "${HOME}/.cache/.aws-sdk/%{__cargo_target}/release"
+%__cargo_outdir_aws_sdk_fips "${HOME}/.cache/.fips/.aws-sdk/%{__cargo_target}/release"
 %__cargo_outdir_static "${HOME}/.cache/.static/%{__cargo_target_static}/release"
+%__cargo_outdir_static_fips "${HOME}/.cache/.fips/.static/%{__cargo_target_static}/release"
 %__cargo_cross_pkg_config PKG_CONFIG_PATH="%{_cross_pkgconfigdir}" PKG_CONFIG_ALLOW_CROSS=1
-%__cargo_cross_env %{__cargo_env} %{__cargo_cross_pkg_config} TARGET_CC="%{_cross_triple}-gnu-gcc"
-%__cargo_cross_env_static %{__cargo_env_static} %{__cargo_cross_pkg_config} TARGET_CC="%{_cross_triple}-musl-gcc"
+
+%__cargo_cross_env_base \
+  CMAKE_TOOLCHAIN_FILE="${PWD}/%{_cross_cmake_toolchain_conf}" ; \
+  export CMAKE_TOOLCHAIN_FILE ; \
+  BINDGEN_EXTRA_CLANG_ARGS="--target=%{_cross_target} --sysroot=%{_cross_rootdir}" ; \
+  export BINDGEN_EXTRA_CLANG_ARGS ; \
+  %{__cargo_cross_pkg_config}
```

Comment:
Are you confident the `"${PWD}"` here is doing the right thing? It seems like it would depend on when the macro was evaluated.

It might be better to explicitly write the cmake toolchain configs to `%{_builddir}` so we can use that as a canonical directory, like we do for `%{_builddir}/.cargo/config`.